.github/workflows/integration_tests.yml

```diff
@@ -18,13 +18,13 @@ jobs:
           sudo rm -rf /usr/local/lib/android
           sudo rm -rf /usr/share/dotnet
           df -h
-      - uses: actions/checkout@v4
+      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
         with:
           submodules: true
-      - uses: dtolnay/rust-toolchain@stable
+      - uses: dtolnay/rust-toolchain@fcf085fcb4b4b8f63f96906cd713eb52181b5ea4
```

Comment:
By removing comment for "stable", it's difficult to figure out the actual rust toolchain used here. I think we should add "with: toolchain: stable" as done in line 140 in this PR.

Reply:
Thanks! I have updated the workflows.